plugins/bcc-login/includes/class-bcc-coreapi-client.php

```diff
@@ -80,6 +80,54 @@ function fetch_groups_for_user(string $user_uid): array
         return $body->data->groupUids;
     }
 
+    public function ensure_subscription_to_person_updates() {
+        if(str_contains(site_url(), "localhost")){
+            return;
+        }
+
+        $subscribed = get_transient("subscribed_to_person_updates");
+        if ($subscribed) {
+            return;
+        }
+
+        $lock = new ExclusiveLock( "subscribe_person_updates" );
+
+        if( $lock->lock( ) == FALSE ){
+            return;
+        }
+        $this->subscribe_to_person_updates();
```

Comment:
Should check transient again here. Otherwise we're just queuing the requests instead of ensuring at most one call.